runner/runner/celery.py

```diff
@@ -11,8 +11,17 @@
 from celery import Celery
 from celery.signals import setup_logging
 
-REDIS_HOST = os.getenv("REDIS_HOST", "localhost")
-REDIS_PORT = os.getenv("REDIS_PORT", "6379")
+broker_workdir = os.getenv("BROKER_WORKDIR", "/tmp")
+
+# Make subdirectory for filesystem broker
+try:
```

Comment:
Doing the directory creation here is problematic in that it makes the mere act of importing from this file something that has side effects.  This is almost always undesirable and frequently confusing.  So for instance, let's say I do the following:

```python
from runner.celery import app
```

Just by doing that, I've created a directory somewhere, assuming I have access to the `BROKER_WORKDIR`.  And if I don't have access, I get the PermissionError informing me that it failed to create the directory, at which point I'm like "geez I was just trying to import something".

There are various ways to refactor this that would avoid this problem.  An easy one might be to just have this portion of the code assume that the BROKER_WORKDIR is something that exists and is accessible.  Then have the main process code, the `runner.py` handle the check and possible creation that you have here.  Maybe just add a `setup_broker` method that gets called prior to calling `spawn_listener` and `spawn_worker`.  That's one option of many.  The point is just to avoid this import side-effect thing.

Reply:
Migrated the broker directory creation to `runner.py`.

I originally put the broker directory creation in the `celery.py` since I figured if you were ever going to import the app, you would want to ensure that the broker directory existed. 

I left the default parent directory of broker as `/tmp`, since it is r/w for everyone by default.